atlp-rwanda/destructors-ec-bn
# feat register user with passport strategy  and jwt 184639721

#### What does this PR do?

This pull request adds a new register user feature to the e-commerce website using Passport strategy with JWT authentication.

#### Description of Task to be completed?

I added a new API endpoint for registering users, which uses Passport and JWT to securely store and transmit user information. I also created a new database schema to store user data, and added validation checks to ensure that user inputs are correctly formatted and meet the site's password strength requirements.

#### How should this be manually tested?

To test the register user feature, follow these steps:

* navigate to the postman and select **/POST** as http method

* add your locally url ex: 
`"http://localhost:<PORT>/api/v1/users/register"`

* select body as json  and apply 

`{
    "firstname": "<user firstname>",
    "lastname": "<user lastname>",
    "email": "<user lastname>",
    "password": "<user password>"
}`

Ensure that you receive a status code which is equal to `201` to ensure that your user are correctly stored in the database.
Try to register the same email to check if you will receive `409` as status code which means that you're Trying to create a resource that already exists

#### Any background context you want to provide?
 The client are requested that user passwords meet specific strength requirements to improve security.

#### What are the relevant pivotal tracker/Trello stories?
* `tracker: 184639721`

#### Screenshots (if appropriate)
* no

#### Questions:

* no Qestion
